pyega3/libs/data_file.py

```diff
@@ -104,15 +105,32 @@ def download_file(self, output_file, num_connections=1):
 
         logging.info(f"Download starting [using {num_connections} connection(s)]...")
 
-        chunk_len = math.ceil(file_size / num_connections)
+        chunk_len = max_slice_size
 
         temporary_directory = os.path.join(os.path.dirname(output_file), ".tmp_download")
+        os.makedirs(temporary_directory, exist_ok=True)
 
         with tqdm(total=int(file_size), unit='B', unit_scale=True) as pbar:
             params = [
-                (temporary_directory, chunk_start_pos, min(chunk_len, file_size - chunk_start_pos), options, pbar)
+                (os.path.join(temporary_directory, self.id), chunk_start_pos,
+                 min(chunk_len, file_size - chunk_start_pos), options, pbar)
                 for chunk_start_pos in range(0, file_size, chunk_len)]
 
+            for file in os.listdir(temporary_directory):
+                match = re.match(r"(.*)-from-(.*)-len-(.*).*", file)
+                file_id = match.group(1)
+                file_from = match.group(2)
+                file_length = match.group(3)
+
+                if file_id != self.id:
+                    continue
+
+                if (file_from, file_length) in [(param[1], param[2]) for param in params]:
+                    continue
+
+                logging.warning(f'Deleting leftover file {file} because the slices sizes is different')
```

Comment:
Maybe `Deleting leftover file {file} because the slice sizes are different`? @malloryfreeberg?